app/decorators/plugins/ecommerce/product_decorator.rb

```diff
@@ -107,8 +108,20 @@ def the_featured_status
     end
   end
 
+  def product_type(variation_id = nil)
+    return get_variation(variation_id).product_type if variation_id.present?
+    return get_default_variation.product_type if is_variation_product?
+    object.get_field_value('ecommerce_product_type').to_s
+  end
+
+  # check if the product is a service
+  def is_service?(variation_id = nil)
+    product_type(variation_id) == 'service_product' ? true : false
```

Comment:
No need for ` product_type(variation_id) == 'service_product' ? true : false`, it already returns true or false with ` product_type(variation_id) == 'service_product'`, doesn't it?